packages/components/src/components/sequence/CallAction.vue

```diff
@@ -230,42 +231,41 @@ export default {
       required: true,
       readonly: true,
     },
-    collapsedActions: {
-      type: Array,
-      required: true,
-    },
-    focusedEvent: {
-      type: Object,
-      default: null,
-    },
     interactive: {
       type: Boolean,
       required: true,
       readonly: true,
     },
-    selected: {
-      type: Boolean,
-    },
     appMap: {
       type: Object,
     },
   },
 
-  data() {
-    return {
-      collapsedActionState: this.collapsedActions,
-    };
-  },
-
   computed: {
+    selected() {
+      return (
+        this.$store.getters.selectedObject &&
+        this.$store.getters.selectedObject instanceof Event &&
+        this.actionSpec.eventIds.includes(this.$store.getters.selectedObject.id)
+      );
+    },
```

Comment:
Does this pattern measurably improve performance? We're still bound by one single observer for each event. What we really need is one observer per event.

Reply:
I originally created `watch` functions on the relevant values in the store and a new `data` property in the component (e.g. `CallAction`) instead of `computed` properties and that had a noticeable performance increase. I changed it to this pattern because the `View in Sequence` button wasn't working if the sequence diagram hadn't been previously viewed/rendered.